src/services/users.ts

```diff
@@ -0,0 +1,37 @@
+'use strict'
+
+import { UsersStorage } from '../storages'
+import { ServiceRegistry } from './serviceRegistry'
+
+export default class UsersService {
+  private constructor (
+    private storage: UsersStorage,
+    private serviceRegistry: ServiceRegistry
+  ) {
+    this.serviceRegistry.register('UsersService', this)
```

Comment:
What if you copy this class creating another service and will forget to change this string?

Reply:
the ServiceRegistry will throw the error: `ServiceRegistry already contains ${name}`. But yes, better to do it automatically